### ISSUE TITLE
Setting eventKinds filter to only .textNote doesn't seem to work?

### ISSUE THREAD

**Issue by ismyhc** [cnixbtc/NostrKit#1]
Hey! Just playing around with nostr and trying to understand how things work.

I wanted to try filter by only EventKind textNote, but im getting other types of data when setting the filter. I see setMetadata and custom types. Is this expected? My assumption would be that setting the filter like below I should only get .textNote events.

Right now I just want to inspect and understand the data.

Here's the snippet Im using.

```swift
let relay = Relay(url: URL(string: "wss://relay.damus.io")!, onEvent: { message in
    switch message {
    case .event(let id, let event):
        print(id)
        print(event)
    case .notice(let notice):
        print(notice)
    case .other(let others):
        print(others)
    }
})

let subscription = Subscription(filters: [
    .init(eventKinds: [.textNote], limit: 10)
])

try? await relay.connect()
try? await relay.subscribe(to: subscription)
```

**Comment by ismyhc** [comment]
Ahhh. It seems that it's the relay I was using. I switched over from wss://relay.damus.io to wss://nostr.bitcoiner.social  and now Im only getting the expected event kinds. I guess I need to check on the client side and filter myself, but it seems the relay should be sending the correct event kinds.

**Comment by ismyhc** [comment]
Closing since I think the issues on the relay side :)

**Comment by ismyhc** [comment]
Re-opening because its seems like the encoding of the EventKind filter is failing. When I print out the actual REQ you can see that there is no filter.

```
["REQ","F6073AB0-4A5A-42FB-A348-0C3229CF7C17",{}] <-- here
```

I think its failing in [ClientMessage](https://github.com/cnixbtc/NostrKit/blob/3d2e45660ae04ac3a2023a13f48d0d2ff3b123b8/Sources/NostrKit/ClientMessage.swift#L18)

```swift
try subscription.filters.forEach { try container.encode($0) }
```

**Comment by cnixbtc** [comment]
Thanks for the detailed report @ismyhc! I'll look into this tomorrow.

**Comment by cnixbtc** [comment]
Seems like I missed a line and forgot to encode the a filter's `eventKinds`. Fixed in https://github.com/cnixbtc/NostrKit/commit/bde474b57a3ae4508a989cead454b44a165f0fcb.

Tagged and released [v0.1.1](https://github.com/cnixbtc/NostrKit/releases/tag/v0.1.1) with the fix. Can you try if that version works for you?

**Comment by ismyhc** [comment]
> Seems like I missed a line and forgot to encode the a filter's `eventKinds`. Fixed in [bde474b](https://github.com/cnixbtc/NostrKit/commit/bde474b57a3ae4508a989cead454b44a165f0fcb).
> 
> Tagged and released [v0.1.1](https://github.com/cnixbtc/NostrKit/releases/tag/v0.1.1) with the fix. Can you try if that version works for you?

Yup! Good to go! :) Thanks for the quick fix!

Building an Apple Multiplatform prototype with NostrKit! If it goes anywhere I will likely open-source it. Will let know if I run into anything else :)
